File: 30DaysOfCode/day10_binaryNumbers/binaryNumbers.pl
#!/usr/bin/perl

$n = <STDIN>;
chomp $n;

$max = 0;
$count = 0;

while ($n > 0) {
    if ($n % 2 == 1) {
        $count++;
        if ($count > $max) {
            $max = $count;
        }
    } else {
        $count = 0;
    }
    
    $n /= 2;
}

print $max;

File: 30DaysOfCode/day10_binaryNumbers/binaryNumbers.rb
#!/bin/ruby

n = gets.strip.to_i
max = 0
count = 0

while n > 0 do
    if n % 2 == 1
        count += 1
        if count > max
            max = count
        end
    else
        count = 0
    end
    
    n /= 2
end
    
puts max
